bluetooth: audio: Update bad code in BASS to be v1.0.1 compliant

gudipudiramanakumar opened the pull request:
Fixes: #86608 
Bad Broadcast Code in BASS/Scan Delegator if BIG_Encryption field value = 0x03 (Bad_Code), Bad_Code shall be set to the value 0xFFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF according to BASS v1.0.1

Thalley commented on subsys/bluetooth/audio/bap_broadcast_assistant.c:
```diff
@@ -280,9 +279,7 @@ static int parse_recv_state(const void *data, uint16_t length,
-		broadcast_code = net_buf_simple_pull_mem(&buf, BT_ISO_BROADCAST_CODE_SIZE);
-		(void)memcpy(recv_state->bad_code, broadcast_code,
-			     sizeof(recv_state->bad_code));
+		memset(recv_state->bad_code, 0xFF, sizeof(recv_state->bad_code));
```
This change should be reverted. The change is for the scan delegator /broadcast sink only

Thalley commented on subsys/bluetooth/audio/bap_scan_delegator.c:
```diff
@@ -949,7 +949,7 @@ static int scan_delegator_broadcast_code(struct bt_conn *conn,
 	}
 
-	broadcast_code = net_buf_simple_pull_mem(buf, sizeof(internal_state->broadcast_code));
+	broadcast_code = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
```
You cannot assign `0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF` to a byte array

Thalley commented on subsys/bluetooth/audio/bap_scan_delegator.c:
```diff
@@ -949,7 +949,7 @@ static int scan_delegator_broadcast_code(struct bt_conn *conn,
 	}
 
-	broadcast_code = net_buf_simple_pull_mem(buf, sizeof(internal_state->broadcast_code));
+	broadcast_code = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
```
This is not where the bad code is assigned. You only need to modify the place where the bad code is assigned

gudipudiramanakumar replied:
Hey, I have changed the requested changes, only changed the bad code assignments and did not completely modify the broadcast_code. Please review and let me know if these changes make sense

Thalley commented on subsys/bluetooth/audio/bap_scan_delegator.c:
```diff
@@ -1368,8 +1368,7 @@ int bt_bap_scan_delegator_set_bis_sync_state(
 	if (internal_state->state.encrypt_state == BT_BAP_BIG_ENC_STATE_BAD_CODE) {
-		(void)memcpy(internal_state->state.bad_code,
-			     internal_state->broadcast_code,
+		(void)memset(internal_state->state.bad_code, 0xFF,
```
Your changes look good in terms of fixing the issue, so based on that I'm OK with approving this. 

One thing that I see as an optimization is that since the bad_code field can only ever be all-0xFF, then we could actually remove the `bad_code` field from the `bt_bap_scan_delegator_recv_state` and then just populate the `read_buf` with 16 * `0xFF` when it is being dumped, read or notified. 

That way we can save 16 octets of RAM per receive state

If that makes sense to you, and you want to update the PR to reflect that, that'd be great, but otherwise we can also keep it simple for now and just focus on fixing the issue.

larsgk replied:
If possible to add to this PR, it would be great - otherwise, maybe a quick followup?

Thalley commented:
@gudipudiramanakumar You need to sign off your commit properly

You also need to modify the test that verifies the bad_code value to expect all-0xFF

gudipudiramanakumar commented:
Sure @Thalley will do the mentioned changes and update the PR

gudipudiramanakumar committed:
bluetooth: audio: Update bad code in BASS to be v1.0.1 compliant

Bad Broadcast Code in BASS/Scan Delegator if BIG_Encryption
field value = 0x03 (Bad_Code), Bad_Code shall be set to the
value 0xFFFFFFFF FFFFFFFF FFFFFFFF FFFFFFFF according to BASS v1.0.1

Signed-off-by: Gudipudi Ramana Kumar <gudipudiramanakumar@gmail.com>

gudipudiramanakumar commented:
Hey @Thalley , sorry for the delay I updated the PR with the mentioned changes.
